### ISSUE TITLE
Feature request: Easier CLI concept regarding release name

### ISSUE THREAD

**Issue by aight8** [kubernetes/helm#2568]
The 'release name' concept is sometimes to inconvenient to use.

```bash
helm install [CHART] [flags] --name <name>
```

```bash
helm upgrade [RELEASE] [CHART] --install --name <name>
```

The problem are the auto generated release. They are mostly none-sense words which give no information about what resource if for.

I think it were easier if (one or more of):
- The package developer can apply a default release name (short name of what the package is about)
- The user can specify a release name in the values file to stick always with the same name

It would be better because:
- On this way you stop accidently release charts because as default it's always the same release
- CLI were easier to use for most use cases

However release names should be reusable by default.
The release name auto generation could be an option '--generate-release-name' - This is much more natural.

**This is not a detailed concept about a solution but describes direction of a potential solution and the issue itself.**

**Comment by technosophos** [comment]
Until 3.0, no changes can be made to these commands that break backward compatibility. (e.g. if a shell script works one way today, it must keep working that way until 3.0.)

That means:

1. Autogenerated names must be the default until 3.0 (though the algorithm for generating could change)
2. If `helm install chartname` is run twice in a row, it MUST produce two releases

So I've marked this for discussion in the 3.x planning phase. But if you can figure out a way around those two requirements, it could be done sooner.

**Comment by TerraTech** [comment]
It would be nice if there was a switch like '-N' to disable the auto-generated names as we tightly control the naming of the releases based on our own charts.  Tonight, I went to install 'grafana' into the monitoring namespace and it ended up being deployed as 'grafana-grafana', where we just want the release and name to both simply be 'grafana'.  There will only ever be one instance in this namespace.  e.g. - this is one of those "I know what I'm doing" kind of switches.  :)

I was hopeful when I saw '--name-template' land, but it didn't provide ability to remove the naming duplication.

If you don't want to expose it as a switch, then an environment variable would be fine, as our helm control is mostly scripted anyhow.

1. switch that disables auto-generation, coupled with the '-n' switch
1. switch that specifies the name *exactly* as given: '-N grafana'
1. developer environment variable that accomplishes <span>#</span>1 above

any of the above would work for our use case.

Thank  you for your consideration.

**Comment by TerraTech** [comment]
@technosophos Any chance my above recommendation would make it into the 3.x series?

**Comment by Vlaaaaaaad** [comment]
+1 for this. The default behavior works, but an extra switch for this would be great. 

See http://blog.lwolf.org/post/how-to-create-ci-cd-pipeline-with-autodeploy-k8s-gitlab-helm/ meanwhile for an alternative when doing CI/CD with helm.

Instead of `if`s I'd like to be able to simply do `helm upgrade --install --values values.yaml --release-name myRelease charts/mychart`

TL;DR:
```
export API_VERSION="$(grep "appVersion" Chart.yaml | cut -d" " -f2)"
export RELEASE_NAME="libr-files-v${API_VERSION/./-}"
export DEPLOYS=$(helm ls | grep $RELEASE_NAME | wc -l)

if [ ${DEPLOYS}  -eq 0 ]; then helm install --name=${RELEASE_NAME} . --namespace=${STAGING_NAMESPACE}; else helm upgrade ${RELEASE_NAME} . --namespace=${STAGING_NAMESPACE}; fi
```

**Comment by colthreepv** [comment]
/remove-lifecycle rotten

Seems a good proposal to me
